lib/pipeline.py

```diff
@@ -280,122 +280,81 @@ def generate_fasta(self, chunks):
-                                                wait_time = 60 - elapsed_time
-                                                if wait_time > 0:
-                                                    time.sleep(wait_time)
-
-                                        arguments = [
-                                            split_fasta_file_path,
-                                            split_iedb_out,
-                                            method,
-                                            a,
-                                            '-r', str(self.iedb_retries),
-                                            '-e', self.iedb_executable,
-                                        ]
-                                        if not isinstance(prediction, IEDBMHCII):
-                                            arguments.extend(['-l', str(epl),])
-                                        lib.call_iedb.main(arguments)
-                                        last_execute_timestamp = datetime.datetime.now()
-                                        p4.print("Making binding predictions on Allele %s and Epitope Length %s with Method %s - Entries %s - Completed" % (a, epl, method, fasta_chunk))
+        argument_sets = []
+        warning_messages = []
+        for (split_start, split_end) in chunks:
+            tsv_chunk = "%d-%d" % (split_start, split_end)
+            fasta_chunk = "%d-%d" % (split_start*2-1, split_end*2)
+            for a in alleles:
+                for epl in epitope_lengths:
+                    if self.input_file_type == 'pvacvector_input_fasta':
+                        split_fasta_file_path = "{}_1-2.{}.tsv".format(self.split_fasta_basename(), epl)
+                    else:
+                        split_fasta_file_path = "%s_%s"%(self.split_fasta_basename(), fasta_chunk)
+                    if os.path.getsize(split_fasta_file_path) == 0:
+                        msg = "Fasta file {} is empty. Skipping".format(split_fasta_file_path)
+                        if msg not in warning_messages:
+                            warning_messages.append(msg)
+                        continue
+                    #begin of per-algorithm processing
+                    for method in prediction_algorithms:
+                        prediction_class = globals()[method]
+                        prediction = prediction_class()
+                        if hasattr(prediction, 'iedb_prediction_method'):
+                            iedb_method = prediction.iedb_prediction_method
+                        else:
+                            iedb_method = method
+                        valid_alleles = prediction.valid_allele_names()
+                        if a not in valid_alleles:
+                            msg = "Allele %s not valid for Method %s. Skipping." % (a, method)
+                            if msg not in warning_messages:
+                                warning_messages.append(msg)
```

Comment:
How big do we anticipate this list of warning messages getting?  It might be useful to use a dictionary for checking if a message exists.